src/fw_update.c

```diff
@@ -264,11 +298,18 @@ static void fw_update_thread(void *arg)
                                                 _main_component->version,
                                                 GOLIOTH_SYS_WAIT_FOREVER);
 
+            golioth_sys_sha256_destroy(download_ctx.sha);
```

Comment:
Can we call `golioth_sys_sha256_destroy()` on line 278 after the call to `golioth_sys_sha256_finish()` so that we only need one call?

Reply:
I see, in a later commit the call to `golioth_sys_sha256_finish()` is moved to another function 👍

Reply:
I have some consternation about this approach because a few lines down there is a call to `fw_update_validate()` and I think this should probably replaced that. I'm trying to limit the scope of this PR so figured we could clean that up in future work.

Reply:
Do you mean that comparing the SHA256 should replace `fw_update_validate()`? I think it's useful to have a point where applications can insert their own custom validation logic.

For my original comment, we can leave it as is given that the organization of the code changed in the later commit.

Reply:
Sorry I wasn't clear. I don't want to replace the call to `fw_update_validate()` but I think it should be grouped with the sha256 verfication. I started to do so with this PR but it becomes a bigger change than I had anticipated. I'll open an issue to track this and we can clean it up (hopefully) as part of the OTA stability epic.

Reply:
Sounds good!